src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -454,37 +478,38 @@ func (c controller) PostInstallConfigs(ctx context.Context, wg *sync.WaitGroup)
 
 	errMessage := ""
 	err, data := c.postInstallConfigs(ctx)
-	// context was cancelled, requires usage of WaitForPredicateWithContext
-	// no reason to set error
-	if ctx.Err() != nil {
-		return
-	}
+	// Even if the context was cancelled, the error is sent to the assisted service.
```

Comment:
Something i not fully understand, if context was cancelled it means that service ended installation process so why we need to send anything to it?

Reply:
I think we still want to send the result of what was done until now.
Before context was cancelled only if timeout occurs.  Now even in case of timeout, context may be cancelled which will stop the controller.

Reply:
Still not sure i understand. Context will be cancelled only in case service finished the installation and there is no point to actually send anything to service as it is done installation process. 
And actually nothing changed in logic here, this verification was added in order not to send error in case service decided to finish the installation and not to confuse with error from controller. I don't think we need to change this behavior

Reply:
With this change, context might also be cancelled in case of timeout expiration.

Reply:
Ok, so we get timeout that means that cluster stopped installation. I am not sure i understand why we cancel main context on timeout and not on cluster state as timeout should mean "failed or degraded" cluster state, no?